src/tools/ApiChecker/ScaffoldTest.cs

```diff
@@ -0,0 +1,149 @@
+            globalNamespace.Imports.Add(new CodeNamespaceImport("Xunit"));
+            globalNamespace.Imports.Add(new CodeNamespaceImport("Xunit.Abstractions"));
+
+            targetUnit.Namespaces.Add(globalNamespace);
+
+            var codeNamespace = new CodeNamespace(@namespace);
+            var targetClass = new CodeTypeDeclaration(@class)
+            {
+                IsClass = true,
+                TypeAttributes = System.Reflection.TypeAttributes.Public | System.Reflection.TypeAttributes.Sealed,
+            };
+
+            targetClass.BaseTypes.Add(new CodeTypeReference("PlaywrightSharpBrowserBaseTest"));
+
+            ///
+            targetClass.CustomAttributes.Add(new CodeAttributeDeclaration("Collection",
+            new CodeAttributeArgument[] {
+                new CodeAttributeArgument(new CodeFieldReferenceExpression(new CodeTypeReferenceExpression("TestConstants"), "TestFixtureBrowserCollectionName"))
+            }));
+
+            targetClass.Comments.Add(new CodeCommentStatement($"<playwright-file>{fileOrigin}</playwright-file>", true));
+            codeNamespace.Types.Add(targetClass);
+
+            targetUnit.Namespaces.Add(codeNamespace);
+
+            // add constructor
+            var constructor = new CodeConstructor()
+            {
+                Attributes = MemberAttributes.Public | MemberAttributes.Final
+            };
+
+            constructor.Parameters.Add(new CodeParameterDeclarationExpression("ITestOutputHelper", "output"));
+            constructor.BaseConstructorArgs.Add(new CodeVariableReferenceExpression("output"));
+            constructor.Comments.Add(new CodeCommentStatement("<inheritdoc/>", true));
+            
+            targetClass.Members.Add(constructor);
+
+            return targetUnit;
+        }
+
+        private static void AddTest(CodeCompileUnit @class, string testDescribe, string testOrigin)
+        {
+            // make name out of the describe
+            string name = _textInfo.ToTitleCase(testDescribe).Replace(" ", String.Empty);
+            Console.WriteLine($"Adding {name}");
+
+            CodeMemberMethod method = new CodeMemberMethod()
+            {
+                Attributes = MemberAttributes.Public | MemberAttributes.Final,
+                ReturnType = new CodeTypeReference("async Task"),
+                Name = name
+            };
+
+            @class.Namespaces[1].Types[0].Members.Add(method);
+
+            method.Comments.Add(new CodeCommentStatement($"<playwright-file>{testOrigin}</playwright-file>", true));
+            method.Comments.Add(new CodeCommentStatement($"<playwright-it>{testDescribe}</playwright-it>", true));
+            method.CustomAttributes.Add(new CodeAttributeDeclaration("Fact",
+            new CodeAttributeArgument[] {
+                new CodeAttributeArgument("Timeout", new CodeFieldReferenceExpression(new CodeTypeReferenceExpression("PlaywrightSharp.Playwright"), "DefaultTimeout"))
```

Comment:
Maybe create the test with the `Skip` attribute? So we could create tests, push them, and work on those in other PRs.